transformation/voltage-normalization/README.md

```diff
@@ -0,0 +1,224 @@
+# Normalize Voltage Levels
+This cookbook is about normalizing voltage levels for energy monitoring.
+
+### Use-case
+> In an energy monitoring system, voltage readings are collected from various sensors and devices. These sensors may use different voltage scales. The analytics services require voltage data normalized to a standardized scale for consistent analysis.
+
+For this use-case, a script, a policy, and a schema are required.
+
+### Schema
+
+To begin with, consider the following plain JSON schema (according to [JSON Schema](https://json-schema.org/)):
+
+`schema-from-sensor.json`:
+```json
+{
+  "type": "object",
+  "properties": {
+    "voltage": {
+      "type": "number"
+    },
+    "sensor_type": {
+      "type": "integer",
+      "enum": [1, 2, 3]
+    },
+    "timestamp": {
+      "type": "number"
+    }
+  },
+  "required": [
+    "voltage",
+    "sensor_type",
+    "timestamp"
+  ]
+}
+```
+which contains two required fields `voltage` and `timestamp` to be of type `number`. The `sensor_type` field of type `integer` can have values 1, 2, or 3, representing different types of sensors reporting in Volts, Millivolts, or Kilovolts, respectively.
+
+After transformation, the field `voltage` is reported in Volts. The schema for transformed MQTT messages is defined as follows:
+
+`schema-for-analytics.json`:
+```json
+{
+  "type": "object",
+  "properties": {
+    "voltage": {
+      "type": "number"
+    },
+    "timestamp": {
+      "type": "number"
+    }
+  },
+  "required": [
+    "voltage",
+    "timestamp"
+  ]
+}
+```
+
+To upload the schemas to the broker, run the following commands:
+
+```bash
+mqtt hivemq schema create --id schema-from-sensor --type json --file schema-from-sensor.json
+mqtt hivemq schema create --id schema-for-analytics --type json --file schema-for-analytics.json
+```
+
+### Transformation function
+In HiveMQ Data Hub from version 4.23 we support JavaScript to transform incoming MQTT
+messages.
+
+The following script normalizes voltage readings from Volts, Millivolts, and Kilovolts to Volts:
+
+```javascript
+function normalizeVoltage(originalVoltage, sensorType) {
+    switch (sensorType) {
+        case 1:
+            return originalVoltage; // Volts
+        case 2:
+            return originalVoltage / 1000; // Millivolts
+        case 3:
+            return originalVoltage * 1000; // Kilovolts
+        default:
+            return originalVoltage; // Default to Volts if unknown sensor type
+    }
+}
+
+function transform(publish, context) {
+    const sensorType = publish.payload.sensor_type;
+
+    publish.payload = {
+        "voltage": normalizeVoltage(publish.payload.voltage, sensorType),
+        "timestamp": publish.payload.timestamp
+    }
```

Comment:
```suggestion
    publish.payload = {
        "voltage": normalizeVoltage(publish.payload.voltage, sensorType),
        "timestamp": publish.payload.timestamp,
    };
```

Reply:
Why so? Other cookbooks (°F to °C for example) do not have it. @busybeetree

Reply:
Oh, I thought I already made that correction for the celsius example.

It doesn't matter, it's just slightly more idiomatic JavaScript, especially the semicolon.